tools/jit/templates/aten_dispatch.h

```diff
@@ -7,15 +7,52 @@
 
 namespace torch { namespace jit {
 
-struct TensorOp {
-  using op_type = std::function<void(const std::vector<at::Tensor> &, std::vector<at::Tensor> &)>;
 
-  TensorOp(op_type op, std::string name, size_t num_inputs)
+// a list of refcounted pointers. this is not called refcounted_list because
+// that sounds like a refcounted list of pointers, which is something else.
+using list_of_retainable = std::vector<at::Retainable*>;
```

Comment:
I think `temp_retainable_list` would be pretty clear. `list_of_retainable` doesn't imply that they won't be freed when it's destructed.